<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OpenLIFUProtocolConfig</class>
 <widget class="qMRMLWidget" name="OpenLIFUProtocolConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>669</width>
    <height>817</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="protocolConfigLayout">
   <item>
    <widget class="ctkComboBox" name="protocolSelector"/>
   </item>
   <item>
    <layout class="QHBoxLayout" name="protocolButtonsLayout">
     <item>
      <widget class="QPushButton" name="loadProtocolFromFileButton">
       <property name="toolTip">
        <string>Load an openlifu protocol from json</string>
       </property>
       <property name="text">
        <string>Load Protocol From File</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="loadProtocolFromDatabaseButton">
       <property name="text">
        <string>Load Protocol From Database</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QWidget" name="permissionsWidget1" native="true">
     <property name="slicer.openlifu.allowed-roles" stdset="0">
      <stringlist>
       <string>admin</string>
      </stringlist>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="createNewProtocolButton">
        <property name="toolTip">
         <string>Create a new openlifu protocol</string>
        </property>
        <property name="text">
         <string>Create New Protocol</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="protocolEditorSectionGroupBox">
        <property name="title">
         <string>Protocol editor</string>
        </property>
        <layout class="QFormLayout" name="protocolEditorSectionLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="protocolNameLabel">
           <property name="toolTip">
            <string>The name of the protocol</string>
           </property>
           <property name="text">
            <string>Name</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="protocolNameLineEdit">
           <property name="toolTip">
            <string>The name of the protocol</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="protocolIdLabel">
           <property name="toolTip">
            <string>The unique identifier of the protocol</string>
           </property>
           <property name="text">
            <string>Protocol ID</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="protocolIdLineEdit">
           <property name="toolTip">
            <string>The unique identifier of the protocol</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="protocolDescriptionLabel">
           <property name="toolTip">
            <string>A more detailed description of the protocol</string>
           </property>
           <property name="text">
            <string>Description</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPlainTextEdit" name="protocolDescriptionTextEdit">
           <property name="toolTip">
            <string>A more detailed description of the protocol</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="pulseIntervalLabel">
           <property name="toolTip">
            <string>Interval (s) between pulses in the sequence</string>
           </property>
           <property name="text">
            <string>Pulse interval (s)</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="pulseIntervalSpinBox">
           <property name="toolTip">
            <string>Interval (s) between pulses in the sequence</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="pulseCountLabel">
           <property name="toolTip">
            <string>Number of pulses in the sequence</string>
           </property>
           <property name="text">
            <string>Pulse count</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="pulseCountSpinBox">
           <property name="toolTip">
            <string>Number of pulses in the sequence</string>
           </property>
           <property name="decimals">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="pulseTrainIntervalLabel">
           <property name="toolTip">
            <string>Interval (s) between pulse trains in the sequence</string>
           </property>
           <property name="text">
            <string>Pulse train interval (s)</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QDoubleSpinBox" name="pulseTrainIntervalSpinBox">
           <property name="toolTip">
            <string>Interval (s) between pulse trains in the sequence</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="pulseTrainCountLabel">
           <property name="toolTip">
            <string>Number of pulse trains in the sequence</string>
           </property>
           <property name="text">
            <string>Pulse train count</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QDoubleSpinBox" name="pulseTrainCountSpinBox">
           <property name="toolTip">
            <string>Number of pulse trains in the sequence</string>
           </property>
           <property name="decimals">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="focalPatternLabel">
           <property name="toolTip">
            <string>The focal pattern used in the protocol. By default, a single point is used</string>
           </property>
           <property name="text">
            <string>Focal pattern type</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QComboBox" name="focalPatternComboBox">
           <property name="toolTip">
            <string>The focal pattern used in the protocol. By default, a single point is used</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="focalPatternOptionsLabel">
           <property name="toolTip">
            <string>Focal pattern options</string>
           </property>
           <property name="text">
            <string>Focal pattern options</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QStackedWidget" name="focalPatternOptionsStackedWidget">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="noFocalPatternPage"/>
           <widget class="QWidget" name="singlePointPage">
            <layout class="QVBoxLayout" name="singlePointOptionsLayout">
             <item>
              <widget class="QLabel" name="singlePointOptionsLabel">
               <property name="text">
                <string>
                                  (No options for the single point pattern)
                                </string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="wheelPage">
            <layout class="QFormLayout" name="wheelOptionsLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="wheelCenterLabel">
               <property name="text">
                <string>Center?</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QCheckBox" name="wheelCenterCheckBox">
               <property name="toolTip">
                <string>
                                  Whether to include the center point of the wheel
                                  pattern
                                </string>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="numSpokesLabel">
               <property name="text">
                <string>Spokes</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="numSpokesSpinBox"/>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="spokeRadiusLabel">
               <property name="text">
                <string>Spoke Radius</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QDoubleSpinBox" name="spokeRadiusSpinBox"/>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="pulseDefinitionWidgetPlaceholder" native="true">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="pulseDefinitionWidgetPlaceholderLabel">
           <property name="text">
            <string>Placeholder for an
OpenLIFUPulseDefinitionWidget</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="saveStateLabel">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="protocolSaveDeleteButtonsLayout">
        <item>
         <widget class="QPushButton" name="protocolFileSaveButton">
          <property name="text">
           <string>Save Protocol To File</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="protocolDatabaseSaveButton">
          <property name="text">
           <string>Save Protocol To Database</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="protocolDatabaseDeleteButton">
          <property name="text">
           <string>Delete Protocol From Database</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="protocolEditRevertDiscardButton">
        <property name="text">
         <string>Edit Protocol</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkComboBox</class>
   <extends>QComboBox</extends>
   <header>ctkComboBox.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
